src/router/Router.js

```diff
@@ -164,6 +167,18 @@ export default class Router extends DisposableBase {
             return this._observeEventsUsingConventions(modelId, object, methodPrefix);
         }
     }
+    addOnErrorHandler(handler) {
+        this._onErrorHandlers.push(handler);
+    }
+    removeOnErrorHandler(handler) {
+        let index = this._onErrorHandlers.indexOf(handler);
+        if(index >= 0) {
+            delete this._onErrorHandlers[index];
+        } else {
+            throw new Error('Unknown error handler.');
```

Comment:
should we really throw here?
It's always one of those questions with remove calls, if it's not there then all you were doing was removing it anyway, but on the other hand if you silently carry on it eludes to a bug in caller code. 

So in summery this is fine :)